airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py

```diff
@@ -0,0 +1,323 @@
+        return self._legacy_stream.availability_strategy
+
+    def check_availability(self, logger: logging.Logger, source: Optional[Source] = None) -> Tuple[bool, Optional[str]]:
+        return self.availability_strategy.check_availability(self._legacy_stream, self._logger, None)
+
+    def get_parser(self) -> FileTypeParser:
+        return self._legacy_stream.get_parser()
+
+    @property
+    def config(self) -> FileBasedStreamConfig:
+        return self._legacy_stream.config
+
+    def get_files(self) -> Iterable[RemoteFile]:
+        return self._legacy_stream.get_files()
+
+    @property
+    def catalog_schema(self) -> Optional[Mapping[str, Any]]:
+        return self._legacy_stream.catalog_schema
+
+    @property
+    def validation_policy(self) -> AbstractSchemaValidationPolicy:
+        return self._legacy_stream.validation_policy
+
+    @cache
+    def get_json_schema(self) -> Mapping[str, Any]:
+        return self._legacy_stream.get_json_schema()
+
+    @property
+    def supports_incremental(self) -> bool:
+        return self._legacy_stream.supports_incremental
+
+    @property
+    def primary_key(self) -> Optional[Union[str, List[str], List[List[str]]]]:
+        return self._legacy_stream.config.primary_key or self.get_parser().get_parser_defined_primary_key(self._legacy_stream.config)
+
+    def as_airbyte_stream(self) -> AirbyteStream:
+        return self._abstract_stream.as_airbyte_stream()
+
+    def read_full_refresh(
+        self,
+        cursor_field: Optional[List[str]],
+        logger: logging.Logger,
+        slice_logger: SliceLogger,
+    ) -> Iterable[StreamData]:
+        """
+        Read full refresh. Delegate to the underlying AbstractStream, ignoring all the parameters
+        :param cursor_field: (ignored)
+        :param logger: (ignored)
+        :param slice_logger: (ignored)
+        :return: Iterable of StreamData
+        """
+        yield from self._read_records()
+
+    def read_incremental(
+        self,
+        cursor_field: Optional[List[str]],
+        logger: logging.Logger,
+        slice_logger: SliceLogger,
+        stream_state: MutableMapping[str, Any],
+        state_manager: ConnectorStateManager,
+        per_stream_state_enabled: bool,
+        internal_config: InternalConfig,
+    ) -> Iterable[StreamData]:
+        yield from self._read_records()
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: Optional[List[str]] = None,
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        stream_state: Optional[Mapping[str, Any]] = None,
+    ) -> Iterable[StreamData]:
+        try:
+            yield from self._read_records()
+        except Exception as exc:
+            if hasattr(self._cursor, "state"):
+                state = str(self._cursor.state)
+            else:
+                # This shouldn't happen if the ConcurrentCursor was used
+                state = "unknown; no state attribute was available on the cursor"
```

Comment:
should this fail loudly?

Reply:
This was copied over from the non-file-based adapters but is no longer needed now that we're inheriting from `AbstractFileBasedStream`.